cmake/AddCxxFlag.cmake

```diff
@@ -44,9 +53,12 @@ function(check_and_add_cxx_flags FLAGS_TO_CHECK)
     ERROR_VARIABLE ERROR_FROM_COMPILATION
     OUTPUT_QUIET)
 
+  add_library(${TARGET_NAME} INTERFACE)
   if(${RESULT} EQUAL 0)
     string(REPLACE ";" " " FLAGS_WITH_SPACES "${FLAGS_TO_CHECK}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLAGS_WITH_SPACES}" PARENT_SCOPE)
+    set_property(TARGET ${TARGET_NAME}
+      APPEND PROPERTY
+      INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:${FLAGS_TO_CHECK}>)
```

Comment:
I think you can delete line 58 that sets the now unused `FLAGS_WITH_SPACES `

Reply:
Done